formver.compiler-plugin/uniqueness/src/org/jetbrains/kotlin/formver/uniqueness/UniqueCheckVisitor.kt

```diff
@@ -19,70 +19,91 @@ import org.jetbrains.kotlin.fir.symbols.impl.FirFunctionSymbol
 import org.jetbrains.kotlin.fir.visitors.FirVisitor
 import org.jetbrains.kotlin.text
 
-object UniqueCheckVisitor : FirVisitor<UniqueLevel, UniqueCheckerContext>() {
-    override fun visitElement(element: FirElement, data: UniqueCheckerContext): UniqueLevel = UniqueLevel.Shared
+data class UniqueVisit(val computedLevel: UniqueLevel, val embedding: UniqueLevelEmbedding?)
 
-    override fun visitSimpleFunction(simpleFunction: FirSimpleFunction, data: UniqueCheckerContext): UniqueLevel {
+object UniqueCheckVisitor : FirVisitor<UniqueVisit, UniqueCheckerContext>() {
+    override fun visitElement(element: FirElement, data: UniqueCheckerContext) =
+        UniqueVisit(UniqueLevel.Shared, null)
```

Comment:
I would prefer to avoid such default behaviour, it results in surprises when you add test cases that should be handled differently but aren't.